node/pkg/ethereum/watcher.go

```diff
@@ -296,26 +302,47 @@ func (e *Watcher) Run(ctx context.Context) error {
 						tx, err := c.TransactionReceipt(timeout, pLock.message.TxHash)
 						cancel()
 						if err != nil && err != rpc.ErrNoResult {
-							logger.Warn("transaction could not be fetched", zap.Stringer("tx", pLock.message.TxHash),
-								zap.Stringer("block", ev.Number), zap.String("eth_network", e.networkName))
+							logger.Warn("transaction could not be fetched",
+								zap.Stringer("tx", pLock.message.TxHash),
+								zap.Stringer("blockhash", key.BlockHash),
+								zap.Stringer("emitter_address", key.EmitterAddress),
+								zap.Uint64("sequence", key.Sequence),
+								zap.Stringer("current_block", ev.Number),
+								zap.String("eth_network", e.networkName),
+								zap.Error(err))
 							continue
 						}
 						if tx == nil {
-							logger.Info("tx was orphaned", zap.Stringer("tx", pLock.message.TxHash),
-								zap.Stringer("block", ev.Number), zap.String("eth_network", e.networkName))
+							logger.Info("tx was orphaned",
+								zap.Stringer("tx", pLock.message.TxHash),
+								zap.Stringer("blockhash", key.BlockHash),
+								zap.Stringer("emitter_address", key.EmitterAddress),
+								zap.Uint64("sequence", key.Sequence),
+								zap.Stringer("current_block", ev.Number),
+								zap.String("eth_network", e.networkName))
 							delete(e.pending, key)
-							ethMessagesOrphaned.WithLabelValues(e.networkName).Inc()
+							ethMessagesOrphaned.WithLabelValues(e.networkName, "not_found").Inc()
 							continue
 						}
 						if tx.BlockHash != key.BlockHash {
-							logger.Info("tx got dropped and mined in a later block; the message should have been reobserved", zap.Stringer("tx", pLock.message.TxHash),
-								zap.Stringer("block", ev.Number), zap.String("eth_network", e.networkName))
+							logger.Info("tx got dropped and mined in a different block; the message should have been reobserved",
+								zap.Stringer("tx", pLock.message.TxHash),
```

Comment:
should also log the actual blockhash

Reply:
<img width="529" alt="image" src="https://user-images.githubusercontent.com/859697/148792756-39ed4a39-5776-4d6e-877b-bd29815988d2.png">

Which of these?

Reply:
Done